checked_message_bench_test.go

```diff
@@ -0,0 +1,96 @@
+package zap_test
```

Comment:
any reason why this is in a separate package? i think `json_encoder_bench_test.go` and `null_encoder_bench_test.go` are both in the `zap` package.

For the logger benchmarks, they are good examples of user interaction so it's nice to have them in a separate package and import `zap`, but I think this should just be part of the `zap` package since it's internal benchmarking